src/schemathesis/runner.py

```diff
@@ -59,11 +85,20 @@ def get_base_url(uri: str) -> str:
     return urlunsplit(parts)
 
 
-def single_test(case: Case, session: requests.Session, url: str, checks: Iterable[Callable]) -> None:
+def single_test(
+    case: Case, session: requests.Session, url: str, checks: Iterable[Callable], stats: StatsCollector
+) -> None:
     """A single test body that will be executed against the target."""
     response = get_response(session, url, case)
+
     for check in checks:
-        check(response)
+        check_name = check.__name__
+        try:
+            check(response)
+            stats.increment(check_name)
+        except AssertionError as e:
+            stats.increment(check_name, error=e)
+            raise e
```

Comment:
Probably we can suppress it until all checks are executed and then reraise some "group" exception, otherwise checks after the first failing one will not be executed. What do you think?

Reply:
Yep, makes sense.